tree/30_generic_methods/package_absent.cf

```diff
@@ -0,0 +1,145 @@
+      "pass2" expression => "pass1";
+      "pass3" expression => "pass2";
+
+  packages:
+
+    debian.architecure_specified.version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => apt_get,
+          architecture   => "${architecture}",
+          version        => "${version}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    debian.architecure_specified.!version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => apt_get,
+          architecture   => "${architecture}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    debian.!architecure_specified.version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => apt_get,
+          version        => "${version}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    debian.!architecure_specified.!version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => apt_get,
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    redhat.architecure_specified.version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => yum,
+          architecture   => "${architecture}",
+          version        => "${version}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    redhat.architecure_specified.!version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => yum,
+          architecture   => "${architecture}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    redhat.!architecure_specified.version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => yum,
+          version        => "${version}",
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+    redhat.!architecure_specified.!version_specified::
+      "${name}"
+          policy         => "absent",
+          package_module => yum,
+          classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}");
+
+  methods:
+
+    (debian|redhat)::
+      "reports" usebundle => _log("Absence of package ${name}${architecture_description}${version_description}", "${old_class_prefix}", "${class_prefix}", @{args});
+
+    !(debian|redhat)::
+      "force failure class" usebundle => _classes_failure("${class_prefix}");
+      "force failure class" usebundle => _classes_failure("${old_class_prefix}");
+      "report failure"      usebundle => _log("Absence of package ${name} - systems other than Debian-, and Red Hat-like are not currently supported", "${old_class_prefix}", "${class_prefix}", @{args});
```

Comment:
I think we should but it is not yet supported by masterfiles, so we have to write our own script for zypper.